File: myledger-h5/src/utils/format.js
export function formatMoney(n) {
  const v = Number(n)
  if (Number.isNaN(v)) return '—'
  return new Intl.NumberFormat('zh-CN', {
    style: 'currency',
    currency: 'CNY',
    minimumFractionDigits: 2,
    maximumFractionDigits: 2,
  }).format(v)
}

export function currentYearMonth(d = new Date()) {
  const y = d.getFullYear()
  const m = String(d.getMonth() + 1).padStart(2, '0')
  return `${y}-${m}`
}

export function shiftYearMonth(ym, delta) {
  const [y, m] = ym.split('-').map(Number)
  const dt = new Date(y, m - 1 + delta, 1)
  return currentYearMonth(dt)
}

/** 表头用：`2026-04` → `26-04`（紧凑、等宽友好） */
export function formatYmCompact(ym) {
  const [y, m] = String(ym || '').split('-')
  if (!y || !m) return String(ym || '')
  return `${y.slice(2)}-${m}`
}

/** `2026-04` → `2026年4月` */
export function formatYearMonthLabel(ym) {
  const parts = String(ym || '').split('-')
  if (parts.length < 2) return ym || '—'
  const y = parts[0]
  const mo = Number(parts[1])
  if (!y || Number.isNaN(mo)) return ym || '—'
  return `${y}年${mo}月`
}

export function formatDateDisplay(s) {
  if (!s) return ''
  return String(s).slice(0, 10)
}

/** 用于统计表：2026-04-03 → 4月3日 */
export function formatMonthDayLabel(s) {
  const d = String(s || '').slice(0, 10)
  if (d.length < 10) return d || '—'
  const [, m, day] = d.split('-')
  return `${Number(m)}月${Number(day)}日`
}

File: myledger-android/app/src/main/java/com/myledger/app/domain/Format.kt
package com.myledger.app.domain

import java.text.NumberFormat
import java.util.Calendar
import java.util.Locale

private val cny: NumberFormat = NumberFormat.getCurrencyInstance(Locale.CHINA).apply {
    minimumFractionDigits = 2
    maximumFractionDigits = 2
}

fun formatMoney(n: Double?): String {
    if (n == null || n.isNaN()) return "—"
    return cny.format(n)
}

fun currentYearMonth(): String {
    val c = Calendar.getInstance()
    val y = c.get(Calendar.YEAR)
    val m = c.get(Calendar.MONTH) + 1
    return String.format(Locale.US, "%04d-%02d", y, m)
}

fun shiftYearMonth(ym: String, delta: Int): String {
    val p = ym.split("-")
    if (p.size < 2) return ym
    val y = p[0].toIntOrNull() ?: return ym
    val mo = p[1].toIntOrNull() ?: return ym
    val c = Calendar.getInstance()
    c.set(Calendar.YEAR, y)
    c.set(Calendar.MONTH, mo - 1 + delta)
    c.set(Calendar.DAY_OF_MONTH, 1)
    return String.format(Locale.US, "%04d-%02d", c.get(Calendar.YEAR), c.get(Calendar.MONTH) + 1)
}

fun formatDateDisplay(s: String?): String {
    if (s.isNullOrBlank()) return ""
    return s.take(10)
}

fun formatYearMonthLabel(ym: String): String {
    val p = ym.split("-")
    if (p.size < 2) return ym
    val y = p[0]
    val mo = p[1].toIntOrNull() ?: return ym
    return "${y}年${mo}月"
}

fun monthLabelFromYm(ym: String): String {
    val mo = ym.split("-").getOrNull(1)?.toIntOrNull()
    return if (mo != null) "${mo}月" else ym
}

fun monthsOfCalendarYear(year: Int): List<String> {
    val y = String.format(Locale.US, "%04d", year)
    return (1..12).map { m -> String.format(Locale.US, "%s-%02d", y, m) }
}

fun visibleMonthsForStats(year: Int): List<String> {
    val full = monthsOfCalendarYear(year)
    val c = Calendar.getInstance()
    val cy = c.get(Calendar.YEAR)
    val cm = c.get(Calendar.MONTH) + 1
    return when {
        year < cy -> full
        year > cy -> full
        else -> full.take(cm)
    }
}
